### ISSUE TITLE
PAX tar format is not enforced for all files added to the final image

### ISSUE THREAD

**Issue by masterzen** [moby/tool#218]
The moby/tool#216 PR makes sure created tar files are using PAX format (because Go 1.10 changed the default format to USTAR instead which can't accomodate long filenames), but when processing the incoming image tar, it blindly reuses whatever format the original header format was for a given file.
This is dangerous because either with links or realname the `prefix` is added and we might cross the USTAR limit (if the incoming file was USTAR encoded), generating an error when adding the file to the archive, as seen here:
 
~~~
DEBU[0046] image tar: docker.elastic.co/elasticsearch/elasticsearch:5.6.10 containers/services/elasticsearch/lower/ add usr/lib/python2.7/site-packages/chardet/charsetgroupprober.pyc
DEBU[0046] image tar: docker.elastic.co/elasticsearch/elasticsearch:5.6.10 containers/services/elasticsearch/lower/ add usr/lib/python2.7/site-packages/chardet/charsetgroupprober.pyo
DEBU[0046] image tar: archive/tar: cannot encode header: Format specifies USTAR; and USTAR cannot encode Linkname="containers/services/elasticsearch/lower/usr/lib/python2.7/site-packages/chardet/charsetgroupprober.pyc" write header error
DEBU[0046] docker rm: 34428aae2600a6cdff5e6c6466177fb4c3671ae38f701f8f95757f0d691f788f
DEBU[0046] docker rm: 34428aae2600a6cdff5e6c6466177fb4c3671ae38f701f8f95757f0d691f788f...Done
FATA[0046] Failed to extract root filesystem for docker.elastic.co/elasticsearch/elasticsearch:5.6.10: archive/tar: cannot encode header: Format specifies USTAR; and USTAR cannot encode Linkname="containers/services/elasticsearch/lower/usr/lib/python2.7/site-packages/chardet/charsetgroupprober.pyc"
~~~

**Comment by rn** [comment]
Thanks for the report. Could you open an issue on https://github.com/linuxkit/linuxkit/ instead? We already moved the code back there and would like to track it there. Thanks

**Comment by masterzen** [comment]
@rn this is a followup of linuxkit/linuxkit#3091 which wasn't completely fixed by moby/tool#216. I added a comment there about the issue, but if you think a new issue is needed, please let me know.

I'm going to submit the PR to the `linuxkit/linuxkit` project then, feel free to close moby/tool#219.

**Comment by masterzen** [comment]
PR submitted as linuxkit/linuxkit#3141.
